### ISSUE TITLE
feat: support jest injectGlobals option to be set to false

### ISSUE THREAD

**Issue by erwinheitzman** [ionic-team/stencil#4006]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/.github/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already include this feature request, without success.

### Describe the Feature Request

I would like to be able to disable the injection of Jest's globals.

### Describe the Use Case

I have tried to run Jest with the injectGlobals option set to false as it is a bad practice to use globals but this did not work.

When trying this I did get it working when removing/commenting this line: https://github.com/ionic-team/stencil/blob/main/src/testing/jest/jest-setup-test-framework.ts#L65

Could we instead use either of the following instead so that this can be supported?
https://jestjs.io/docs/configuration#testtimeout-number
https://jestjs.io/docs/cli#--testtimeoutnumber

I did try to import jest in the `jest-setup-test-framework.ts` file but this lead to an error that jest cannot be used before initializing the test environment so it might be good to move setting the timeout anyway.

This seems like a relatively small change so I hope it is something that has little impact 👍

**Comment by alicewriteswrongs** [comment]
Hey @erwinheitzman thanks for filing this issue! I believe I understand the feature request here but just to make sure: the problem you're running into is that setting [`injectGlobals`](https://jestjs.io/docs/cli#--injectglobals) to `false` isn't working because of this line:

https://github.com/ionic-team/stencil/blob/024afda8ec248fb300d8b87a399ad2b888a35780/src/testing/jest/jest-setup-test-framework.ts#L65

Presumably if Jest is set to not inject globals then the `jest` global won't be defined and you'll get an `Uncaught ReferenceError: jest is not defined` there. Is that all correct?

If that is, I'd also like to get more of a sense of why you'd like to use the option. The normal practice when using Jest is to allow it to inject globals (like `describe`, `test`, etc) which are then used as a sort of pseudo-DSL in test files. While I normally don't like to rely on global variables (especially in JS!) this sort of pattern is generally the norm with Jest and similar test runners (mocha, etc). Anyway, I'm just curious to get more of a sense of what you're trying and unable to do right now.

Thanks again for reporting!

**Comment by erwinheitzman** [comment]
Hi @alicewriteswrongs thank you for the fast reply and you are absolutely correct!

The reason we want to use this option is because we are using a tool called WebdriverIO which is an end to end test framework that under the hood uses either Mocha, Jasmine or Cucumber (you can choose which) as a testrunner. The problem is that Jest and for example Mocha, both have the `describe` `it` and `expect` globals and are causing Typescript to see a duplicate.

Luckily in the latest version of WebdriverIO we can now import the globals manually which resolves half of the problem where the other half would be Jests's globals which we now also want to import manually.

I hope this paints a good picture of why we would like to see this Jest option supported.

**Comment by alicewriteswrongs** [comment]
@erwinheitzman yes thanks for breaking that down, it makes sense what you're trying to do and why Stencil's Jest setup isn't working for your use-case right now.

I'm going to label this as a feature request to gauge community interest. 

Also just for some additional context, the Stencil team is looking at some issues related to our Jest setup and testing libraries during the next quarter, so we'll be taking a closer look at various things related to this soon.

Thanks for filing the issue!

**Comment by erwinheitzman** [comment]
@alicewriteswrongs will these discussions be held in a public call by any chance? Since I'm an experienced test engineer and am currently working with Stencil, I could perhaps be of help.

**Comment by alicewriteswrongs** [comment]
Hey @erwinheitzman we're still sorting out internally where our priorities are and doing some research / 'ideating' around what we want Stencil's testing future to look like. At present we don't have much concrete stuff to share - although I can say that this remains a priority for our (small) team, and also that if we are planning big changes we will likely go through a public RFC once the team has a solid plan worked out. I'll do my best to remember to ping this issue with anything along those lines!
